File: CarRentalBackEnd/src/main/java/com/carRental/service/PaymentService.java
package com.carRental.service;

import java.util.List;
import com.carRental.dto.PaymentDTO;
import com.carRental.dto.PaymentOrderDTO;

// Service interface for processing payments
public interface PaymentService {

    // Processes a payment order and saves it
    PaymentDTO makePayment(PaymentOrderDTO paymentOrderDTO);

    // Retrieves payment details by ID
    PaymentDTO getPaymentById(Long id);

    // Retrieves all payment records
    List<PaymentDTO> getAllPayments();

    // Retrieves payment details specific to a booking
    PaymentDTO getPaymentByBookingId(Long bookingId);
}

File: CarRentalFrontEnd/src/services/paymentService.js
import api from './api';

// Retrieves payment details associated with a specific booking
export const getPaymentByBookingId = async (bookingId) => {
    try {
        const response = await api.get(`/payments/booking/${bookingId}`);
        return response.data;
    } catch (error) {
        throw error.response?.data || error.message;
    }
};
